contracts/strategy/RWAStrategy.sol

```diff
@@ -0,0 +1,145 @@
+        // pass a loss on to the user withdrawing.
+        asset.safeTransferFrom(managerAddress, address(this), _amount);
+        totalInvestedInRWA -= _amount;
+    }
+
+    /**
+     * @dev Optional function for a strategist to override that will
+     * allow management to manually withdraw deployed funds from the
+     * yield source if a strategy is shutdown.
+     *
+     * This should attempt to free `_amount`, noting that `_amount` may
+     * be more than is currently deployed.
+     *
+     * NOTE: This will not realize any profits or losses. A separate
+     * {report} will be needed in order to record any profit/loss. If
+     * a report may need to be called after a shutdown it is important
+     * to check if the strategy is shutdown during {_harvestAndReport}
+     * so that it does not simply re-deploy all funds that had been freed.
+     *
+     * EX:
+     *   if(freeAsset > 0 && !TokenizedStrategy.isShutdown()) {
+     *       depositFunds...
+     *    }
+     *
+     * @param _amount The amount of asset to attempt to free.
+     */
+    function _emergencyWithdraw(uint256 _amount) internal override {
+        uint256 amountToWithdraw = Math.min(_amount, totalInvestedInRWA);
+        asset.safeTransferFrom(managerAddress, address(this), amountToWithdraw);
+        asset.transfer(TokenizedStrategy.management(), amountToWithdraw);
+        totalInvestedInRWA -= amountToWithdraw;
+    }
+
+    function setMinAmount(uint256 _minAmount) external onlyManagement {
+        minAmount = _minAmount;
+    }
+
+    /// @notice Allows the manager to report gains or losses.
+    /// @dev Should be called before calling report() to report the amount of the gain or loss.
+    /// @dev The manager can only report gains or losses.
+    /// @param _gain The amount of the gain.
+    /// @param _loss The amount of the loss.
+    function reportGainOrLoss(uint256 _gain, uint256 _loss) external {
+        require(msg.sender == managerAddress, "Only the manager can report gains or losses");
+
+        if (_gain > 0) {
+            require(_loss == 0, "Cannot report both gain and loss");
+            // Transfer the gain from the manager to the strategy contract
+            ERC20(asset).safeTransferFrom(managerAddress, address(this), _gain);
+            totalGains += _gain;
+        } else if (_loss > 0) {
+            require(_loss <= totalInvestedInRWA, "Cannot report loss more than total invested");
+            totalLosses += _loss;
+            totalInvestedInRWA -= _loss;
+        }
+    }
+
+    /// @notice Allows the manager to set the deposit limit.
+    /// @param _depositLimit The new deposit limit.
+    function setDepositLimit(uint256 _depositLimit) external onlyManagement {
+        require(_depositLimit > 0, "Deposit limit must be greater than 0.");
+        require(_depositLimit > TokenizedStrategy.totalIdle() + totalInvested, "Deposit limit must be greater than total invested.");
```

Comment:
There is no `totalInvested` variable in this contract, I suppose I tried to use `totalInvestedInRWA` let's rename it to "totalInvested" cause "totalInvestedInRWA" name has redundancy.

Reply:
and please compile contracts and run tests before committing changes